examples/investigate_streams.rs

```diff
@@ -0,0 +1,266 @@
+                .takes_value(false)
+                .help("If specified, compress files with bzip2 for comparison."),
+            Arg::with_name("lzma")
+                .long("--compare-to-lzma")
+                .takes_value(false)
+                .help("If specified, compress files with lzma for comparison."),
+        ])
+        .get_matches();
+
+    let root = matches.value_of("path").unwrap(); // Checked by clap.
+    let glob = format!("{}/**/*", root); // Quick and dirty walk through subdirectories.
+    let options = Options {
+        compare_to_bzip: matches.is_present("bzip"),
+        compare_to_lzma: matches.is_present("lzma"),
+    };
+
+    let mut number_of_files = 0;
+    let mut stats = HashMap::new();
+    for path in
+        glob::glob(&glob).unwrap_or_else(|e| panic!("Cannot walk directory {}: {:?}", root, e))
+    {
+        let path = path.expect("Error while walking");
+        eprintln!("Visiting {:?}", path);
+        let meta_data = std::fs::metadata(&path).unwrap();
+        if meta_data.is_dir() {
+            continue;
+        }
+
+        number_of_files += 1;
+
+        let extension = path.extension().unwrap().to_str().unwrap().to_string();
+        match extension.as_str() {
+            "binjs" | "js" | "entropy" => {
+                // binjs_decode doesn't output a brotli-compressed version of these
+                // files, so we call `brotli` manually to obtain one and evaluate
+                // the file size.
+                let brotli_size = get_brotli_size(&path);
+
+                let name = match extension.as_str() {
+                    // For `binjs` and `js` files, we only care about the extension.
+                    "binjs" | "js" => extension.clone(),
+                    _ => path.file_name().unwrap().to_str().unwrap().to_string(),
+                };
+
+                let info = stats
+                    .entry(extension.clone())
+                    .or_insert_with(|| HashMap::new())
+                    .entry(name)
+                    .or_insert_with(|| Compressions::default());
+                info.raw += meta_data.len();
+                info.brotli += brotli_size as u64;
+
+                if options.compare_to_bzip {
+                    info.bzip += get_bzip2_size(&path);
+                };
+                if options.compare_to_lzma {
+                    info.lzma += get_lzma_size(&path);
+                };
+            }
+            // Brotli files, output by binjs_encode.
+            // Remove the `bro` from the name, store in `Compressions::brotli`.
+            "bro" => {
+                let path = path.as_path().with_extension("");
+                let name = path.file_name().unwrap().to_str().unwrap().to_string();
+                let extension = path.extension().unwrap().to_str().unwrap().to_string();
+                let info = stats
+                    .entry(extension.clone())
+                    .or_insert_with(|| HashMap::new())
```

Comment:
Nit: you can use `.or_default()` for all these instances.

Reply:
Actually, it requires type annotations if I replace everything with `or_default()`. I figure it's not worth the tradeoff, what do you think?

Reply:
It might require one annotation for the `stats` where it's defined, but the call sites should be simpler.

Reply:
Ok, will do.